src/backend/base/langflow/components/vectorstores/CassandraGraph.py

```diff
@@ -0,0 +1,244 @@
+            required=True,
+        ),
+        MessageTextInput(
+            name="table_name",
+            display_name="Table Name",
+            info="The name of the table (or AstraDB collection) where vectors will be stored.",
+            required=True,
+        ),
+        DropdownInput(
+            name="setup_mode",
+            display_name="Setup Mode",
+            info="Configuration mode for setting up the Cassandra table, with options like 'Sync' or 'Off'.",
+            options=["Sync", "Off"],
+            value="Sync",
+            advanced=True,
+        ),
+        DictInput(
+            name="cluster_kwargs",
+            display_name="Cluster arguments",
+            info="Optional dictionary of additional keyword arguments for the Cassandra cluster.",
+            advanced=True,
+            is_list=True,
+        ),
+        MultilineInput(name="search_query", display_name="Search Query"),
+        DataInput(
+            name="ingest_data",
+            display_name="Ingest Data",
+            is_list=True,
+        ),
+        HandleInput(name="embedding", display_name="Embedding", input_types=["Embeddings"]),
+        IntInput(
+            name="number_of_results",
+            display_name="Number of Results",
+            info="Number of results to return.",
+            value=4,
+            advanced=True,
+        ),
+        DropdownInput(
+            name="search_type",
+            display_name="Search Type",
+            info="Search type to use",
+            options=["Traversal", "MMR traversal", "Similarity", "Similarity with score threshold", "MMR (Max Marginal Relevance)"],
+            value="Traversal",
+            advanced=True,
+        ),
+        IntInput(
+            name="depth",
+            display_name="Depth of traversal",
+            info="The maximum depth of edges to traverse. (when using 'Traversal' or 'MMR traversal')",
+            value=1,
+            advanced=True,
+        ),
+        FloatInput(
+            name="search_score_threshold",
+            display_name="Search Score Threshold",
+            info="Minimum similarity score threshold for search results. (when using 'Similarity with score threshold')",
+            value=0,
+            advanced=True,
+        ),
+        DictInput(
+            name="search_filter",
+            display_name="Search Metadata Filter",
+            info="Optional dictionary of filters to apply to the search query.",
+            advanced=True,
+            is_list=True,
+        ),
+    ]
+
+    @check_cached_vector_store
+    def build_vector_store(self) -> CassandraGraphVectorStore:
+        try:
+            import cassio
+            from langchain_community.utilities.cassandra import SetupMode
+        except ImportError:
+            raise ImportError(
+                "Could not import cassio integration package. " "Please install it with `pip install cassio`."
+            )
+
+        from uuid import UUID
```

Comment:
Maybe could be a global import, if only just for the sanity check that uuid is available at component load time?

Reply:
`uuid` is built-in so it will always be available.
I don't mind making it a global import. This part of code was copy/pasted from `CassandraVectorStoreComponent`.

Reply:
done